### PR TITLE
Ft 12250: Added prefect task-runs to cli

### LINKED ISSUE #12250: Create `prefect task-runs ...` commands to list and view metadata about task runs
Similar to how users can view metadata about flow runs via `prefect flow-runs ...` commands, we should also support `prefect task-runs ...` commands to view the same for task runs.

### PR DESCRIPTION
**Pull request by JiginJayaprakash**
closes #12250 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/prefect/cli/task_run.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                exit_with_error(f"Task run {id!r} not found!")
+            else:
+                raise
+
+    app.console.print(Pretty(task_run))
+
+
+@task_run_app.command()
+async def ls(
+    task_name: List[str] = typer.Option(None, help="Name of the task"),
```
Comment (chrisguidry) [inline src/prefect/cli/task_run.py:53]: How about `task_run_name` here?

**Review by chrisguidry — APPROVED** [review]
Very nice, thank you for the contribution!

File: src/prefect/cli/task_run.py (hunk 1 above)
Comment (JiginJayaprakash) [reply src/prefect/cli/task_run.py]: Thank you , I have updated as per your comments

**Comment by serinamarie** [comment]
@JiginJayaprakash If you wanted to add some tests for this in a follow-up PR that would be pretty sweet!

**Pull request merged**